torchvision/models/detection/_utils.py

```diff
@@ -468,3 +468,31 @@ def retrieve_out_channels(model: nn.Module, size: Tuple[int, int]) -> List[int]:
         model.train()
 
     return out_channels
+
+
+@torch.jit.unused
+def _onnx_tracing_topk_min(input: Tensor, orig_kval: int, axis: int) -> Tuple[int, int]:
+    """
+    ONNX spec requires the k-value to be less than or equal to the number of inputs along
+    provided dim. Certain models uses python's min() function to determine whether to use
+    the provided k-value or the specified dim axis value.
+
+    However in cases where the model is being exported in tracing mode, python min() is
+    static causing the model to be traced incorrectly and eventually fail at the topk node.
+    In order to avoid this situation, in tracing mode, torch.min() is used instead.
+
+    Args:
+        input (Tensor): The orignal input tensor.
+        orig_kval (int): The provided k-value.
+        axis(int): Axis along which we retreive the input size.
+
+    Returns:
+        (Tuple[int, int]): Input size along axis, appropriately selected k-value.
+    """
+    from torch.onnx import operators
+
+    axis_dim_val = operators.shape_as_tensor(input)[axis].unsqueeze(0)
+    min_kval = torch.min(torch.cat((torch.tensor([orig_kval], dtype=axis_dim_val.dtype), axis_dim_val), 0))
+
+    # for mypy we cast at runtime
+    return cast(int, axis_dim_val), cast(int, min_kval)
```

Comment:
Casting a tensor to an int doesn't look very clean to me. Can't you use [item](https://pytorch.org/docs/stable/generated/torch.Tensor.item.html)?

Reply:
Hi @prabhat00155 , 

Seems that the `item()` will cause something wrong in the tracing mode, and there is a similar workaround in the detection transformation:

https://github.com/pytorch/vision/blob/44e93beb0833ff51ffe75d994fe660434335b53d/torchvision/models/detection/transform.py#L47-L50

Reply:
Yup, as @zhiqwang mentioned, tensor.item() is replaced with a constant during tracing.
This is a snippet from https://pytorch.org/docs/stable/onnx.html that addresses this limitation.
```And rather than using [torch.Tensor.item()] (which converts a Tensor to a Python built-in number):
# Bad! y.item() will be replaced with a constant during tracing.
```